Firestore/core/src/firebase/firestore/immutable/sorted_set.h

```diff
@@ -74,6 +74,23 @@ class SortedSet : public SortedContainer {
     return SortedSet{map_.insert(key, {})};
   }
 
+  ABSL_MUST_USE_RESULT SortedSet union_with(const SortedSet& other) const {
+    auto result_ptr = const_cast<SortedSet*>(this);
```

Comment:
`const_cast` is something to generally avoid. It's sometimes required when interoperating with broken C APIs but is something we should otherwise not use. Why not:

```c++
const SortedSet* result_ptr = this;
const SortedSet* other_ptr = &other;

if (...) {
  // swap
}

SortedSet result = *result_ptr;
...
```